src/effects/effectknobparameterslot.cpp

```diff
@@ -97,7 +83,8 @@ void EffectKnobParameterSlot::loadManifestParameter(unsigned int iParameterNumbe
         //          .arg(dValue).arg(dMinimum).arg(dMinimumLimit).arg(dMaximum).arg(dMaximumLimit).arg(dDefault);
 
         EffectManifestParameter::ControlHint type = m_pManifestParameter->controlHint();
-        m_pControlValue->setBehaviour(type, dMinimum, dMaximum);
+        ControlEffectKnob* pControlValue = static_cast<ControlEffectKnob*>(m_pControlValue);
```

Comment:
This casting shouldn't be needed if you have a ControlEffectKnob member variable.

Reply:
Then we won't be able to use the ControlObject objects in EffectParameterSlotBase. Or wait, not if we can have virtual member variables also.